### PR TITLE
Add missing worktree_dir check and test case

### LINKED ISSUE #5280: git_worktree_validate return true if the worktree dir not exists.
I add a worktree to my repo. After remove the worktree path from my finder, the `git_worktree_validate` still return 0.

I check the code in the `worktree.c`:

https://github.com/libgit2/libgit2/blob/b246bed5ab83035d8aef95f1b7ff10dd746db7cb/src/worktree.c#L235-L262

It seems that it does not check the `worktree_path` exists. It check the `gitdir_path`、`commondir_path` and `parent_path`, but these paths are the origin git path, it exists.

### PR DESCRIPTION
**Pull request by rbmclean**
Fixes #5280

### CODE REVIEW

File: tests/worktree/worktree.c [hunk 1]
```diff
@@ -610,3 +610,17 @@ void test_worktree_worktree__foreach_worktree_lists_all_worktrees(void)
...
 	cl_git_pass(git_repository_foreach_worktree(fixture.repo, foreach_worktree_cb, &counter));
 }
+
+void test_worktree_worktree__validate_invalid_worktreedir(void)
+{
+	git_worktree *wt;
+
+	cl_git_pass(git_worktree_lookup(&wt, fixture.repo, "testrepo-worktree"));
+	git__free(wt->worktree_path);
+	wt->worktree_path = "/path/to/invalid/worktreedir";
```
Comment (ethomson) [inline tests/worktree/worktree.c:620]: This feels a little bit like too much magic to me.  😀  Is there a way that we can invalidate the work tree?  If I’m reading this correctly, we can just rm the work tree itself which would serve the same purpose but test the intent more directly and without having to change the innards of the work tree object.

**Comment by ethomson** [comment]
Thanks for the fix and the test!  Wonder if we can tweak the test to be a little more direct but otherwise 👍

**Comment by rbmclean** [comment]
My understanding of clar isn't great, but I didn't want to delete the worktree as I suspected it would cause issues. Now it's renamed and seems to still work as intended. I ran the test suite twice in a row on my machine to make sure it was renamed back correctly.

**Comment by rbmclean** [comment]
Is there anything else that needs to be done for this?

File: tests/worktree/worktree.c [hunk 2]
```diff
@@ -610,3 +610,14 @@ void test_worktree_worktree__foreach_worktree_lists_all_worktrees(void)
...
 	cl_git_pass(git_repository_foreach_worktree(fixture.repo, foreach_worktree_cb, &counter));
 }
+
+void test_worktree_worktree__validate_invalid_worktreedir(void)
+{
+	git_worktree *wt;
+
+	cl_git_pass(git_worktree_lookup(&wt, fixture.repo, "testrepo-worktree"));
+	p_rename("testrepo-worktree", "testrepo-worktree-tmp");
+	cl_git_fail(git_worktree_validate(wt));
```
Comment (ethomson) [inline tests/worktree/worktree.c:620]:
```suggestion
	cl_git_fail(git_worktree_validate(wt));
	p_rename("testrepo-worktree-tmp", "testrepo-worktree");
```

File: src/worktree.c [hunk 3]
```diff
@@ -259,7 +259,14 @@ int git_worktree_validate(const git_worktree *wt)
 			wt->commondir_path);
 		return GIT_ERROR;
 	}
-
+	
+	if (!git_path_exists(wt->worktree_path)) {
+		git_error_set(GIT_ERROR_WORKTREE,
+			"worktree directory ('%s') does not exist ",
```
Comment (ethomson) [inline src/worktree.c:265]:
```suggestion
			"worktree directory '%s' does not exist",
```

**Review by ethomson — COMMENTED** [review]
This looks good to me - I would move the cleanup rename into the test itself, and omit the trailing space in the error message, but otherwise this looks good.

**Comment by ethomson** [comment]
Since the changes that I'm suggesting are super minor and stylistic, I'm going to go ahead and commit the suggestions that I made.

Thanks again for the PR!

**Review by ethomson — APPROVED** [review]

**Pull request merged**